test/WebJobs.Script.Tests/Scale/FunctionsScaleManagerTests.cs

```diff
@@ -301,11 +313,64 @@ public void GetScaleInstancesToProcess_Returns_Expected(bool targetBaseScalingEn
+            FunctionsScaleManager manager = new FunctionsScaleManager(scaleMonitorManagerMock.Object, _metricsRepositoryMock.Object,
+                targetScalerManagerMock.Object, _concurrencyStatusRepositoryMock.Object, functionsHostingConfigurationMock.Object,
+                env, _loggerFactory);
+
+            manager.GetScalersToSample(out List<IScaleMonitor> scaleMonitorsToProcess, out List<ITargetScaler> targetScalesToProcess);
 
             Assert.Equal(scaleMonitorsToProcess.Count(), expectedScaleMonitorCount);
             Assert.Equal(targetScalesToProcess.Count(), expectedTargetScalerCount);
         }
+
+        [Fact]
+        public async Task GetScalerSamples_Fallbacks_ToScaleMonitor()
+        {
+            List<IScaleMonitor> scaleMonitors = new List<IScaleMonitor>
+            {
+                new TestScaleMonitor<ScaleMetrics>("function1-test-test", "function1"),
+            };
+            Mock<IScaleMonitorManager> scaleMonitorManagerMock = new Mock<IScaleMonitorManager>(MockBehavior.Strict);
+            scaleMonitorManagerMock.Setup(x => x.GetMonitors()).Returns(scaleMonitors);
+
+            List<ITargetScaler> targetScalers = new List<ITargetScaler>
+            {
+                new FaultyTargetScaler()
+                {
+                    TargetScalerDescriptor = new TargetScalerDescriptor("function1")
+                },
+                new TestTargetScaler2()
+                {
+                    TargetScalerDescriptor = new TargetScalerDescriptor("function2")
+                }
+            };
+            Mock<ITargetScalerManager> targetScalerManagerMock = new Mock<ITargetScalerManager>(MockBehavior.Strict);
+            targetScalerManagerMock.Setup(x => x.GetTargetScalers()).Returns(targetScalers);
+
+            Mock<IFunctionsHostingConfiguration> functionsHostingConfigurationMock = new Mock<IFunctionsHostingConfiguration>(MockBehavior.Strict);
+            string assemblyName = Assembly.GetExecutingAssembly().GetName().Name;
+            functionsHostingConfigurationMock.Setup(p => p.GetValue(assemblyName, It.IsAny<string>())).Returns("1");
+
+            TestEnvironment env = new TestEnvironment();
+            env.SetEnvironmentVariable(EnvironmentSettingNames.TargetBaseScalingEnabled, "1");
+
+            FunctionsScaleManager manager = new FunctionsScaleManager(scaleMonitorManagerMock.Object, _metricsRepositoryMock.Object,
+                targetScalerManagerMock.Object, _concurrencyStatusRepositoryMock.Object, functionsHostingConfigurationMock.Object,
+                env, _loggerFactory);
+
+            var context = new ScaleStatusContext()
+            {
+                WorkerCount = 1
+            };
+
+            manager.GetScalersToSample(out List<IScaleMonitor> monitors1, out List<ITargetScaler> scalers1);
+            Assert.Equal(monitors1.Count(), 0);
+            Assert.Equal(scalers1.Count(), 2);
+            await manager.GetScaleStatusAsync(context);
```

Comment:
Please also verify the scale status result here and below, that the vote for the faulty scaler is None, and TargetWorkerCount is as expected.